sdk/internal/recording/server.go

```diff
@@ -0,0 +1,329 @@
+		return "test-proxy-standalone-linux-arm64.tar.gz", nil
+	case runtime.GOOS == "darwin" && runtime.GOARCH == "amd64":
+		return "test-proxy-standalone-osx-x64.zip", nil
+	case runtime.GOOS == "darwin" && runtime.GOARCH == "arm64":
+		return "test-proxy-standalone-osx-arm64.zip", nil
+	default:
+		return "", fmt.Errorf("unsupported OS/Arch combination: %s/%s", runtime.GOOS, runtime.GOARCH)
+	}
+}
+
+func extractTestProxyZip(archivePath string, outputDir string) error {
+    // Open the zip file
+    r, err := zip.OpenReader(archivePath)
+    if err != nil {
+        panic(err)
+    }
+    defer r.Close()
+
+    for _, f := range r.File {
+        targetPath := filepath.Join(outputDir, f.Name)
+
+        log.Println("Extracting", targetPath)
+
+        if f.FileInfo().IsDir() {
+            os.MkdirAll(targetPath, f.Mode())
+            continue
+        }
+
+        file, err := os.OpenFile(targetPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+        if err != nil {
+            return err
+        }
+        defer file.Close()
+
+        rc, err := f.Open()
+        if err != nil {
+            return err
+        }
+        defer rc.Close()
+
+        if _, err = io.Copy(file, rc); err != nil {
+            return err
+        }
+    }
+
+	return nil
+}
+
+func extractTestProxyArchive(archivePath string, outputDir string) error {
+	log.Printf("Extracting %s\n", archivePath)
+	file, err := os.Open(archivePath)
+	if err != nil {
+        return err
+    }
+    defer file.Close()
+	gzipReader, err := gzip.NewReader(file)
+    if err != nil {
+        return err
+    }
+    defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+
+	for {
+        header, err := tarReader.Next()
+        if err == io.EOF {
+            break
+        }
+        if err != nil {
+            return err
+        }
+
+        targetPath := filepath.Join(outputDir, header.Name)
+
+        log.Println("Extracting", targetPath)
+
+        switch header.Typeflag {
+        case tar.TypeDir:
+            if err := os.MkdirAll(targetPath, 0755); err != nil {
+                return err
+            }
+        case tar.TypeReg:
+            file, err := os.Create(targetPath)
+            if err != nil {
+                return err
+            }
+            defer file.Close()
+
+            if _, err := io.Copy(file, tarReader); err != nil {
+                return err
+            }
+        default:
+            log.Printf("Unable to extract type %c in file %s\n", header.Typeflag, header.Name)
+        }
+	}
+
+	return nil
+}
+
+func extractTestProxy(archivePath string, outputDir string) error {
+	if strings.HasSuffix(archivePath, ".zip") {
+		return extractTestProxyZip(archivePath, outputDir)
+	} else {
+		return extractTestProxyArchive(archivePath, outputDir)
+	}
+}
+
+func ensureTestProxyInstalled(proxyVersion string, proxyPath string, proxyDir string) error {
+	lockFile := filepath.Join(os.TempDir(), "test-proxy-install.lock")
```

Comment:
It'd be good to print this in case something goes wrong and we have to manually delete it.

Reply:
I print it here: https://github.com/Azure/azure-sdk-for-go/pull/21168/files/6f3a5d3f2127b1b59426da58f59919f0a083eaf0#diff-90f9ae1ef4aacd672ce8f7a44bcc77a6f8550e2321aefe9d1111896bade770d9R169

I tried to get handling for os.Interrupt working as well in case someone hits ctrl-c on the download, but the go test command seems to send that itself so it made things more complicated.

Reply:
I also updated to print the lock file we're waiting on at the start, in case people ctrl-c out thinking it's hung.